orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/PipelineExecutionLauncherSpec.groovy

```diff
@@ -119,4 +124,30 @@ class PipelineExecutionLauncherSpec extends Specification {
     config = [id: "whatever", stages: []]
     json = objectMapper.writeValueAsString(config)
   }
+
+  @Unroll
+  def "sets executionEngine correctly"() {
+    given:
+    @Subject def launcher = create()
+
+    when:
+    launcher.start(PIPELINE, json)
+
+    then:
+    1 * executionRepository.store({
+      it.executionEngine == expected
+    })
+
+    where:
+    supplied                | expected
+    [executionEngine: "v1"] | v3
+    [executionEngine: "v2"] | v3
+    [executionEngine: "v3"] | v3
```

Comment:
Note here that v1 and v2 will map to v3. v1 and v2 are completely obsolete execution engines, but I did notice that some internal pipeline configuration still contain `v2` at least.